### PR TITLE
Add Core Metrics Endpoint, Ranged Queries

### PR DESCRIPTION
**Pull request by mcmcgrath13**
* Add a `/telemetry/{buoyId}/CoreMetrics/{timeRange}` endpoint which aggregates the metrics defined in #24 for any of our time range queries
* Abstract the telemetry endpoints so it's easier to mix/match timeRanges with the tables / metrics
* Make better use of the swagger docs, so that all of the APIs are correctly documented on `/api-docs`

### CODE REVIEW

File: app/routes/erddap/buoy.js [hunk 1]
```diff
@@ -22,35 +22,47 @@ router.param("source", (req, res, next, source) => {
...
- *        in: query
- *        description: End Time
- *        required: true
- *        type: string
- *      - name: numPoints
- *        in: query
- *        description: max number of points to return
- *        required: false
- *        type: integer
+ *       - in: path
```
Comment (mirestrepo) [inline app/routes/erddap/buoy.js:29]: So swagger works? ;)

**Review by mirestrepo — APPROVED** [review]
Looks good to me!

File: app/routes/erddap/buoy.js (hunk 1 above)
Comment (mcmcgrath13) [reply app/routes/erddap/buoy.js:29]: yup - turns out it's been there all along 🙃

File: app/clients/telemetry.js [hunk 2]
```diff
@@ -12,21 +13,46 @@ const pool = mysql.createPool({
 
 const query = util.promisify(pool.query).bind(pool);
 
-const getLatestRecord = async (buoyId, tableType) => {
+const ALL_COLUMNS = ["*"];
+const getColumns = (variables = ALL_COLUMNS) => {
+  // note - this is okay because this code is the only source of variable names - this would
+  // be bad if we allowed user defined variables here
+  return (columns = variables.map((variable) => SqlString.raw(variable)));
+};
```
Comment (broarr) [inline app/clients/telemetry.js:21]: Took me a while to trace that out, but it looks like all the vars are being set through the CORE_METRICS, is that right?

**Review by broarr — APPROVED** [review]
LGTM!

File: app/routes/telemetry/index.js [hunk 3]
```diff
@@ -30,83 +73,152 @@ router.param("buoyId", (req, res, next, buoyId) => {
...
-);
+router.param("range", (req, res, next, range) => {
+  const queryFn = RANGES[range];
+  if (queryFn) {
+    req.queryFn = queryFn(req);
+    next();
+  } else {
+    next(new Error("unknown query range type"));
+  }
+});
```
[the remark this answers is not in the document]
Comment (mcmcgrath13) [reply app/routes/telemetry/index.js:93]: yup! it both provides the function we need to the request object and acts as a guard to make sure we only respond to requests we know what to do with

File: app/clients/telemetry.js (hunk 2 above)
Comment (mcmcgrath13) [reply app/clients/telemetry.js:21]: Either there or the `ALL_COLUMNS` default.  I could see a world where we have some other pre-canned sets of tables/variables, but always defined from within the api.

**Pull request merged**